tool/tsh/tsh.go

```diff
@@ -215,6 +213,10 @@ func Run(args []string, underTest bool) {
 	app.Flag("gops-addr", "Specify gops addr to listen on").Hidden().StringVar(&cf.GopsAddr)
 	app.Flag("skip-version-check", "Skip version checking between server and client.").BoolVar(&cf.SkipVersionCheck)
 	app.Flag("debug", "Verbose logging to stdout").Short('d').BoolVar(&cf.Debug)
+	app.Flag("ignore-local-ssh-agent", "Do not load generated SSH certificates into the local ssh-agent (specified via $SSH_AUTH_SOCK). You can also set TELEPORT_NO_LOCAL_SSH_AGENT environment variable.").
+		Envar(noLocalSSHAgentEnvVar).
+		Default(fmt.Sprint(maybeUsingGPGAgent())). // Note: the default is dynamic, based on our guess of the user environment.
```

Comment:
So, I really like that we're trying to be smart by guessing whether gpg-agent is running and using that to decide whether to add certificates to the agent or not. This is definitely what we should do to make the experience better for people.

However, I am not much of a fan of us dynamically changing the default for a `tsh` flag and its help text based on an external parameter. I can see a potential situation where someone looks at `tsh` help output on a machine where `gpg-agent` is running, says "ok, the default behaviour is to ignore the ssh agent", then runs `tsh` assuming this to be true on a machine where `gpg-agent` isn't running and gets different behaviour.

I'm not 100% sure about a good solution to this though.

Reply:
One note: `tsh --help` doesn't actually show you the defaults.

I went with this approach because I couldn't find a way to check whether a flag is set with `kingpin`. If we could check whether this is set, output could be very clear about us guessing.
Let me know if there's a way!

An alternative is to not do any guessing at all and let the user always set env var or the flag. This may be a niche-enough case to not require smarts in `tsh`.
WDYT?

Reply:
I didn't realise that! If `tsh --help` doesn't show the defaults then I think this way is fine.

We should still (IMO) be explicit when deliberately choosing not to add keys to a `gpg-agent` though. If this isn't possible with `kingpin`, maybe we just remove the automatic behaviour and add a specific note about gpg-agent's lack of certificate support to our docs. 

I believe that most of the people experiencing the gpg-agent problem are already following the linked Github issue so disseminating information about the new flag should be easy.

Reply:
Do we want a short flag for this @benarent?

Reply:
I think it's Ok as a long flag and keep short flags for common workloads. IF we get a lot of feedback of people recreating tsh environments, we can come back and create a short flag.

Reply:
Removed guessing since it makes everything a bit awkward.
We'll just have to mention this in the docs. @benarent I probably shouldn't add it to `docs/4.2/...`. I can update the docs when we have a 4.3 directory.

Reply:
For 4.3 docs just create a draft PR against `docs/4.3-base` see https://github.com/gravitational/teleport/pull/3697 for reference.